### PR TITLE
Examples and Errors for Scripting API

### PR DESCRIPTION
**Pull request by dbendele**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/niveristand-data-sharing-framework-custom-device/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Adds and exports DSF error file with descriptions. Adds and exports examples for the scripting API.

### Why should this Pull Request be merged?

Examples demonstrate use of the scripting API and specific error descriptions help debug scripting applications for the Data Sharing Framework.

### What testing has been done?

Local build successful. PR build will be validated by installation and validation of package. New scripting tests are also being added.

### CODE REVIEW

File: VeriStand/Data Sharing Custom Device/Source/Data Sharing Framework Custom Device.lvproj [hunk 1]
```diff
@@ -21,7 +21,16 @@
 		<Property Name="server.vi.callsEnabled" Type="Bool">true</Property>
 		<Property Name="server.vi.propertiesEnabled" Type="Bool">true</Property>
 		<Property Name="specify.custom.address" Type="Bool">false</Property>
+		<Item Name="Documents" Type="Folder">
+			<Item Name="DSF-errors.txt" Type="Document" URL="../../../../Docs/DSF-errors.txt"/>
+		</Item>
 		<Item Name="Tests" Type="Folder">
+			<Item Name="Scripting" Type="Folder">
+				<Item Name="Examples" Type="Folder">
+					<Item Name="Scripting.lvclass" Type="LVClass" URL="../Tests/Scripting/Add Loopback/Scripting.lvclass"/>
```
Comment (buckd) [inline VeriStand/Data Sharing Custom Device/Source/Data Sharing Framework Custom Device.lvproj:30]: Why is `Scripting.lvclass` in an **Add Loopback** folder on disk? This seems like it should just be under the **Scripting** directory. Same for `targets.ini`.

**Review by buckd — COMMENTED** [review]
- [x] `Scripting.lvclass` has a class data item called **None**, which isn't used. Remove this extraneous field from the class.
- [x] Scripting Examples should be part of a library. It will make things easier for building the examples for distribution.
- [x] What are the values in the case structures for the tests (example below)? If I were to have to debug these tests, I have no idea what these multipliers are doing.
image

File: VeriStand/Data Sharing Custom Device/Source/Data Sharing Framework Custom Device.lvproj (hunk 1 above)
Comment (dbendele) [reply VeriStand/Data Sharing Custom Device/Source/Data Sharing Framework Custom Device.lvproj]: originally going to add two test classes 'Add Loopback' and 'Import Send' but decided to include both tests in the same test class.

Comment (dbendele) [reply VeriStand/Data Sharing Custom Device/Source/Data Sharing Framework Custom Device.lvproj]: https://github.com/ni/niveristand-data-sharing-framework-custom-device/pull/20/commits/023029302fd93e9682001b411fe35f7509b4402f

**Comment by dbendele** [comment]
image

**Review by niphilj — APPROVED** [review]
Thanks for adding the structure comments in the tests.

**Comment by dbendele** [comment]
we are aware of an issue with the use of version-specific packages to install a common file to a shared (non-version-specific) destination. As currently designed, installation of multiple versions of the DSF Custom Device will install a DSF-errors.txt file to 
National Instruments\Shared\Errors\English. Uninstallation of any of the versions removes the error text which removes the error description support for remaining version of the DSF custom device.

The following proposal was submitted by @rtzoeller 
>NIPM does not do ref counting for file-based package files. ... the CD packages install error files to the LV runtime errors directory directly, and the CD scripting APIs depend on a single package that installs the error file to the development errors directory

**Pull request merged**